### PR TITLE
Add tests for successful `virtual_specs` checks

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Comes from https://github.com/conda-forge/miniforge/pull/626. `micromamba` doesn't like something in that dry-run, apparently :)

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](https://github.com/conda/constructor/blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Comment by jaimergp** [comment]
This is weird. This PR makes things at https://github.com/conda-forge/miniforge/pull/626 pass, but it fails here. Previous iterations made things work here, but fail at the miniforge PR. 🤷

File: constructor/nsis/main.nsi.tmpl [hunk 1]
```diff
@@ -1139,7 +1139,7 @@ Section "Install"
     System::Call 'kernel32::SetEnvironmentVariable(t,t)i("CONDA_SOLVER", "classic").r0'
     SetDetailsPrint TextOnly
     DetailPrint "Checking virtual specs..."
-    push '"$INSTDIR\_conda.exe" create --dry-run --prefix "$INSTDIR" --offline @VIRTUAL_SPECS@'
+    push '"$INSTDIR\_conda.exe" create --dry-run --prefix "$INSTDIR\envs\_virtual_specs_checks" --offline @VIRTUAL_SPECS@'
```
Comment (marcoesters) [inline constructor/nsis/main.nsi.tmpl:1142]: Do we need to create a temporary directory here, too?

Comment (jaimergp) [reply constructor/nsis/main.nsi.tmpl:1142]: Micromamba is not used often on Windows due to the lack of menuinst, so it's not as critical. No clue how to create a tempdir via NSIS either.

Comment (marcoesters) [reply constructor/nsis/main.nsi.tmpl:1142]: I think I know why we don't need to. We are setting `CONDA_PKGS_DIRS` here: https://github.com/conda/constructor/blob/main/constructor/nsis/main.nsi.tmpl#L1124

In the `sh` and `pkg` installers, that environment variable is set after the checks for virtual specs.

**Review by marcoesters — APPROVED** [review]

File: news/843-conda-pkgs-dirs-virtual-specs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* <news item>
+
+### Bug fixes
+
+* Fix a couple issues in the `virtual_specs` feature (set `CONDA_PKGS_DIRS` for `micromamba`, do not rely `$BATCH` for PKG, update the Windows example). (#843)
```
Comment (marcoesters) [inline news/843-conda-pkgs-dirs-virtual-specs:7]:
```suggestion
* Fix a couple issues in the `virtual_specs` feature (set `CONDA_PKGS_DIRS` for `micromamba`, do not rely on `$BATCH` for PKG, update the Windows example). (#843)
```

**Review by marcoesters — APPROVED** [review]

**Pull request merged**